### ISSUE TITLE
New bug found in chat

### ISSUE THREAD

**Issue by Christby** [cse110-sp23-group3/cse110-sp23-group3#72]
@henryzhang03 I just found a bug for the chat history. 
When I click New Chat before completing the whole process of palm reading, weird things will happen such that the new chat continues from the state where we left on the last chat.
Here are some examples: 
<img width="1425" alt="Screenshot 2023-06-06 at 21 44 14" src="https://github.com/cse110-sp23-group3/cse110-sp23-group3/assets/97484108/ddadbe5a-79fb-4667-bc29-53f04bdfe34c">
<img width="1435" alt="Screenshot 2023-06-06 at 21 44 27" src="https://github.com/cse110-sp23-group3/cse110-sp23-group3/assets/97484108/d2750d0b-015b-4718-aa4b-f2532b679808">

**Comment by Christby** [comment]
Also I found that there's always one chat history left on the screen no matter how many times I clear the local storage

**Comment by henryzhang03** [comment]
got it, will try to fix in next commit @Christby

**Comment by dolivares11** [comment]
> Also I found that there's always one chat history left on the screen no matter how many times I clear the local storage

Is this a bug because it's showing the user that that is the chat being saved currently?

**Comment by Christby** [comment]
@dolivares11 No, it's a non-empty chat that cannot be deleted. I'm not sure if the problem is on my computer or the code

**Comment by henryzhang03** [comment]
@dolivares11 @Christby don't worry, it's because of how readPalm is structured as an async function due to having to wait for the API calls. I need to change it a little and it will be fine.

**Comment by henryzhang03** [comment]
Not yet, I will fix it soon, I've tried a ton of things but none have worked, but I think I have a solution
